File: webflux-with-mongo-example/src/main/java/com/example/webfluxwithmongoexample/domain/user/service/UserService.java
package com.example.webfluxwithmongoexample.domain.user.service;

import com.example.webfluxwithmongoexample.domain.user.repository.UserRepository;
import com.example.webfluxwithmongoexample.domain.user.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
@RequiredArgsConstructor
public class UserService {
    private final UserRepository userRepository;

    public Mono<UserInfo> save(UserCommand command) {
        User user = new User(command);
        Mono<User> savedUser = userRepository.save(user);
        return savedUser.map(saved -> new UserInfo(saved.getName()));
    }
}

File: warmup-example/src/main/kotlin/com/example/warmupexample/domain/user/service/UserService.kt
package com.example.warmupexample.domain.user.service

import com.example.warmupexample.domain.user.extensions.toInfo
import com.example.warmupexample.domain.user.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(
    private val userRepository: UserRepository,
) {
    fun getUsers(): List<UserInfo> {
        val users = userRepository.getUsers()
        return users.map { user -> user.toInfo() }
    }
}